RFC-0009-jdbc-join-push-down.md

```diff
@@ -0,0 +1,570 @@
+- If not, return the node.
+
+![JdbcJoinPushdown optimizer](RFC-0009-jdbc-join-push-down/After_JdbcJoinPushdown.png)
+
+JdbcJoinPushdown optimizer will create a TableScanNode structure which is able to hold all the jdbc tables which are grouped as part of above implementation. Below is the proposed structure for the new TableScanNode
+
+![Image 2](RFC-0009-jdbc-join-push-down/in-depth-design-image-2.png)
+
+## PredicatePushDown Optimizer
+
+#### Pushdown the overall filter to the newly created TableScanNode.
+After creating Single TableScanNode for grouped tables (refer point 7) we need to pushdown the FilterNode (join criteria specific to the grouped tables of new tableScanNode and all filters specific to the group tables) on connector level for the applicable filter and maintain the FilterNode for presto if it is not able to pushdown. For this we are just invoking predicate pushdown after jdbc join pushdown optimizer and there is no code change.
+
+## JdbcComputePushdown Optimizer
+Using JdbcComputePushdown optimizer, we are pushing down the join criteria as additional predicate. For doing this, we enhanced JdbcComputePushdown optimizer by adding a join predicate to sql translator. 
+```
+private final JdbcJoinPredicateToSqlTranslator jdbcJoinPredicateToSqlTranslator;
+```
+
+```
+this.jdbcJoinPredicateToSqlTranslator = new JdbcJoinPredicateToSqlTranslator(
+                functionMetadataManager,
+                buildFunctionTranslator(ImmutableSet.of(JoinOperatorTranslators.class)),
+                identifierQuote);
+```
+We have also added visitPlan() method and enhanced the visitFilter() method.
+
+Enhancements done to visitFilter() are from this PR : https://github.com/prestodb/presto/pull/16412/
+Before, if all the filter could not be pushed down, nothing would be pushed down.
+The changes help in finding what all can be pushed down and able to be translated. The filters that can't be pushed down are kept in a new FilterNode.
+
+## Changes in SPI module 
+
+- Added 2 new classes : JoinTableInfo, JoinTableSet
+- JoinTableInfo holds the information about a table that is undergoing pushdown. It has details about tableHandle, assignments and outputVariables.
+- JoinTableSet is used to store a set of JoinTableInfo objects.
+- These are explained in detail below.
+
+We have added a new Connector capability called SUPPORTS_JOIN_PUSHDOWN ConnectorCapabilities enum in SPI. Any connector which wants to participate in pushdown need to added this Connector capability to their Set of ConnectorCapabilities. JDBC has done it like [this](#3-use-multijoinnode-to-group-jdbc-tables-based-on-connector-name)
+
+## Edge cases
+
+### Self Join
+Self join is a scenario where a table is joined with itself. This works natively in Presto. When you do a self join in presto, that is sent as select statements to the datasource first. On top of that presto does all the joining and filtering etc. 
```

Comment:
```suggestion
Self join is a scenario where a table is joined with itself. This is handled in Presto by assigning unique TableScanNode's for each instance of the table being joined
```

nit : Please keep all refs to `Presto` with an uppercase-P since its a proper noun